md/bdm-access-control.md

```diff
@@ -314,3 +314,153 @@ Deploy your access control file.
+
+If you preview the page at this point, both jan.fischer and helen.kelly can access your new application, select from a drop down list either *Mathematics* or *Physics*, wich should display all the Mathematics and all the Physics requests, respectively, for both users.
+
+**5. Security implementation**
+
+- Got to *BonitaStudioSubscription-7.7.0/workspace/tomcat/setup/*
+- Run *setup pull*
+- Go the */BonitaStudioSubscription-7.7.0/workspace/tomcat/setup/platform_conf/current/tenants/1/tenant_portal/* folder that should have just appeared.
+- Open the *dynamic-permissions-checks-custom.properties*, and add the following line:
+``` GET|bdm/businessData/com.company.model.Request=[check|SubjectTeacherPermissionRule] ```
+- This line indicates that before executing any queries on the com.company.model.Request object types in the BDM, a verification has to be run. In this case we indicate that the verification is a groovy script, *SubjectTeacherPermissionRule.groovy*, which we will now create.
+- Go to */BonitaStudioSubscription-7.7.0/workspace/tomcat/setup/platform_conf/current/tenants/1/tenant_security_scripts*. Create a file *SubjectTeacherPermissionRule.groovy*, with the following content:
+
+```groovy
+package org.bonitasoft.permissions
+
+import org.bonitasoft.engine.api.APIAccessor
+import org.bonitasoft.engine.api.Logger
+import org.bonitasoft.engine.api.permission.APICallContext
+import org.bonitasoft.engine.api.permission.PermissionRule
+import org.bonitasoft.engine.profile.Profile
+import org.bonitasoft.engine.profile.ProfileCriterion
+import org.bonitasoft.engine.session.APISession
+
+class SubjectTeacherPermissionRule implements PermissionRule {
+
+    @Override
+    boolean isAllowed(APISession apiSession, APICallContext apiCallContext, APIAccessor apiAccessor, Logger logger) {
+        APISession session = apiSession
+        long currentUserId = session.getUserId()
+        List<Profile> profilesForUser = apiAccessor.getProfileAPI().getProfilesForUser(currentUserId, 0, 50, ProfileCriterion.ID_ASC)
+        if (!apiCallContext.getQueryString().contains("q=findBySubject")) {
+            println "q=findBySubject found in query string"
+            return true
+        }
+
+        def filters = apiCallContext.getFilters()
+        if (filters.containsKey("subject")) {
+            def subjectAsString = filters.get("subject")
+            for (Profile p : profilesForUser) {
+                if (p.name.contains(subjectAsString)){
+                    return true
+                }
+            }
+            return false
+        }
+        return true
+    }
+}
+```
+
+- Return to *BonitaStudioSubscription-7.7.0/workspace/tomcat/setup/* and run *setup push*. this will upload the *dynamic-permissions-checks-custom.properties* file to the server.
+- Restart your web server. The new security rule from the *dynamic-permissions-checks-custom.properties* file now active.
```

Comment:
Restart your web server through menu "Server -> Restart Web server" (or whatever is the real name)...
...**is** now active